Document WebGLContextEvent() (#25622)
mdn/content: files/en-us/web/api/webglcontextevent/webglcontextevent/index.md

---
title: WebGLContextEvent()
slug: Web/API/WebGLContextEvent/WebGLContextEvent
page-type: web-api-constructor
browser-compat: api.WebGLContextEvent.WebGLContextEvent
---

{{APIRef("WebGL")}}

The **`WebGLContextEvent()`** constructor creates a new {{domxref("WebGLContextEvent")}} object.

> > **Note:** A web developer doesn't typically need to call this constructor, as the browser creates these objects itself when firing events.

Reviewer: I think we usually use "You" to talk to all readers, not just web developers. Also, I think people would come to this page because they want to trigger a context event manually. For that, there is a dedicated function available, so mention it here.

```suggestion
> **Note:** You typically don't need to call this constructor; the browser creates these objects automatically when WebGL context events get fired. To manually trigger a `webglcontextlost` event, use {{domxref("WEBGL_lose_context.loseContext()")}}.
```

## Syntax

```js-nolint
new WebGLContextEvent(type, options)
```

### Parameters

- `type`

>   - : A string with the name of the event.

Reviewer: ```suggestion
  - : A string indicating the type of the event.
```

>     It is case-sensitive and browsers set it to `webglcontextcreationerror`, `webglcontextlost`, or `weblcontextrestored`.

Reviewer: ```suggestion
    It is case-sensitive and should be one of `webglcontextcreationerror`, `webglcontextlost`, or `weblcontextrestored`.
```

- `options` {{Optional_inline}}

>   - : An object that, _in addition of the properties defined in {{domxref("Event/Event", "Event()")}}_, has the following properties:

Reviewer 2: ```suggestion
  - : An object that, _in addition to the properties defined in {{domxref("Event/Event", "Event()")}}_, has the following properties:
```

>     - `status` {{Optional_inline}}

Reviewer: ```suggestion
    - `statusMessage` {{Optional_inline}}
```

>       - : A string with some additional status information. It defaults to the empty string (`""`).

Reviewer: ```suggestion
      - : A string with additional status information. It defaults to the empty string (`""`).
```
